packages/core/src/external/carequality/ihe-gateway-v2/outbound/xca/create/iti39-envelope.ts

```diff
@@ -22,10 +26,16 @@ export type BulkSignedDR = {
   outboundRequest: OutboundDocumentRetrievalReq;
 };
 
-function wrapDocUniqueIdIfLowercaseUuid(docUniqueId: string): string {
-  const valid = validate(docUniqueId);
-  const lowercase = docUniqueId === docUniqueId.toLowerCase();
-  return valid && lowercase ? wrapIdInUrnUuid(docUniqueId) : docUniqueId;
+/*
+ * Wraps the document unique id in a urn uuid if the gateway requires it and the document unique id is a lowercase UUID.
+ */
+function wrapDocUniqueIdIfLowercaseUuid(docUniqueId: string, gateway: XCAGateway): string {
+  const isValidUuid = validateUuid(docUniqueId);
+  const isLowercase = docUniqueId === docUniqueId.toLowerCase();
+  const isUrnUuidRequired = requiresUrnUuidPrefix(gateway);
```

Comment:
Not to be picky, but this is confusing for me: when I read the `createITI39SoapEnvelope()` function, I see  `"urn:DocumentUniqueId": wrapDocUniqueIdIfLowercaseUuid(...)` which makes me thing we always do it.

Only opening the function I can see that it only does it for some cases - and even then this is buried deep (3rd cmd - could be the first thing, and calls another function).

What if we did something like this:

```ts
type GetDocumentUniqueIdFn = (docUniqueId: string) => string;

/**
 * Function called on 'createITI39SoapEnvelope()'
 */
function getDocumentUniqueId(docUniqueId: string, gateway: XCAGateway): string {
  const getDocUniqueIdFn = getDocumentUniqueIdFunctionByGateway(gateway);
  return getDocUniqueIdFn(docUniqueId);
}

const getDocumentUniqueIdMapping: Record<string, GetDocumentUniqueIdFn> = {
  [surescriptsOid]: wrapDocUniqueIdIfLowercaseUuid,
};

function getDocumentUniqueIdFunctionByGateway(gateway: XCAGateway) {
  const fn = getDocumentUniqueIdMapping[gateway.homeCommunityId];
  if (fn) return fn;
  return defaultGetDocumentUniqueId;
}

function defaultGetDocumentUniqueId(docUniqueId: string): string {
  return docUniqueId;
}

/*
 * Wraps the document unique id in a urn uuid if the gateway requires it and the document unique id is a lowercase UUID.
 */
function wrapDocUniqueIdIfLowercaseUuid(docUniqueId: string): string {
  const isValidUuid = validateUuid(docUniqueId);
  const isLowercase = docUniqueId === docUniqueId.toLowerCase();
  return isValidUuid && isLowercase ? wrapIdInUrnUuid(docUniqueId) : docUniqueId;
}
```